code/obj/item/gun/kinetic.dm

```diff
@@ -2453,3 +2453,9 @@ ABSTRACT_TYPE(/obj/item/gun/survival_rifle_barrel)
 	alter_projectile(obj/projectile/P)
 		. = ..()
 		P.proj_data.shot_sound = 'sound/weapons/sawnoff.ogg'
+
+	on_spin_emote(mob/living/carbon/human/user)
+		if(src.broke_open)
+			src.attack_self(user)
```

Comment:
doc comment on attack_self
![image](https://github.com/goonstation/goonstation/assets/358431/5642b89e-81eb-4a23-bab8-73ab09b0d9cf)
```suggestion
			src.AttackSelf(user)
```
but also a sensible option would be to just put the code for doing this into a separate proc that's called from on_spin_emote and attack_self both.